btk/metrics.py

```diff
@@ -1,484 +1,497 @@
-    detection/deblending/measurement algorithm in meas_generator. Each
-    time the algorithm is called, it is run on a batch of blends yielded
-    by the meas_generator.
-    """
-
-    def get_detections(self):
-        """Returns input blend catalog and detection catalog for
-        the detection performed.
-
-        Returns:
-            Results of the detection algorithm are returned as:
-                true_tables: List of astropy Table of the blend catalogs of the
-                    batch. Length of tables must be the batch size. x and y
-                    coordinate values must be under columns named 'x_peak' and 'y_peak'
-                    respectively, in pixels from bottom left corner as (0, 0).
-                detected_tables: List of astropy Table of output from detection
-                    algorithm. Length of tables must be the batch size. x and y
-                    coordinate values must be under columns named 'x_peak' and 'y_peak'
-                    respectively, in pixels from bottom left corner as (0, 0).
-        """
-        blend_op, deblend_op, _ = next(self.meas_generator)
-        true_tables = blend_op["blend_list"]
-        detected_tables = []
-        for i in range(len(true_tables)):
-            detected_centers = deblend_op[i]["peaks"]
-            detected_table = astropy.table.Table(detected_centers, names=["x_peak", "y_peak"])
-            detected_tables.append(detected_table)
-        return true_tables, detected_tables
-
-
-def make_true_seg_map(image, threshold):
-    """Returns a boolean segmentation map corresponding to pixels in
-    image above a certain threshold value.
     Args:
-        image: Image to estimate segmentation map of
-        threshold: Pixels above this threshold are marked as belonging to
-            segmentation map
-
-    Returns:
-        Boolean segmentation map of the image
+        iso_image (np.array): Array of shape = (H,W) corresponding to image of the isolated
+            galaxy you are calculating blendedness for.
+        blend_iso_images (np.array): Array of shape = (N, H, W) where N is the number of galaxies
+            in the blend and each image of this array corresponds to an isolated galaxy that is
+            part of the blend (includes `iso_image`).
     """
-    seg_map = np.zeros_like(image)
-    seg_map[image < threshold] = 0
-    seg_map[image >= threshold] = 1
-    return seg_map.astype(np.bool)
-
+    num = np.sum(iso_image * iso_image)
+    denom = np.sum(np.sum(blend_iso_images, axis=0) * iso_image)
+    return 1 - num / denom
 
-def get_closest_neighbor_distance(true_table):
-    """Returns a astropy.table.column with the distance to the closest object.
 
-    Function uses scipy.spatial to compute distance between the object centers.
-    If object is the only one in the blend then the `min_dist` value is set to
-    np.inf.
+def meas_ellipticity(image, additional_params, shear_est="KSB"):
```

Comment:
as discussed during the telecon, maybe we should stick to KSB and rename the method `meas_ksb_ellipticity` for instance